src/parser.rs

```diff
@@ -2123,11 +2135,19 @@ impl<'input, 'abbrev, 'unit, Endian> EntriesCursor<'input, 'abbrev, 'unit, Endia
         }
     }
 
+    /// Return the current offset in bytes inside the compilation unit
+    pub fn current_offset(&self) -> usize {
+        let ptr = self.input.as_ptr() as *const u8 as usize;
+        let start_ptr = self.unit.entries_buf.as_ptr() as *const u8 as usize;
+        ptr - start_ptr + self.unit.header_size()
```

Comment:
Not sure how @fitzgen feels but you may want to do checked arithmetic here to handle bogus values in the DWARF without panicking.